commcare_connect/opportunity/visit_import.py

```diff
@@ -277,11 +277,6 @@ def _bulk_update_payments(opportunity: Opportunity, imported_data: Dataset) -> P
     return PaymentImportStatus(seen_users, missing_users)
 
 
-def _cache_key(currency_code, date=None):
-    return [currency_code, date.toordinal() if date else None]
-
-
-@quickcache(vary_on=_cache_key, timeout=12 * 60 * 60)
```

Comment:
One thing is, in this version, the API will be hit once, but in the modified version you are going to hit the same API separately for each currency_code. You might consider pulling the API call out into separate function and cache that.